tests/custom_selector_name.rs

```diff
@@ -0,0 +1,59 @@
+// This test is the same as basic.rs but with custom context selectors
+
+use snafu::{ResultExt, Snafu};
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, Snafu)]
+enum Error {
+    #[snafu(
+        context(OpenConfigContext),
```

Comment:
Seeing it in use, I wonder if we want to give some more "namespacing". For example:

```rust
        context(name(OpenConfigContext)),
```

Or

```rust
        context(selector(OpenConfigContext)),
```

Or

```rust
        context(selector_name(OpenConfigContext)),
```